File: PROGRAMMERS/가운데글자가져오기.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <string>
#include <cstring>
using namespace std;

#define endl '\n'

typedef long long ll;
typedef pair<int, int> pii;
typedef pair<ll, ll> pll;

string solution(string s)
{
	string answer = "";
	int length = (s.length() >> 1);
	if (s.length() & 1) {
		answer = s[length];
	}
	else {
		answer += s[length - 1];
		answer += s[length];
	}
	return answer;
}

int main()
{
	ios_base::sync_with_stdio(false);
	cout.tie(nullptr); cin.tie(nullptr);

	string s = "qwer";
	string ret = solution(s);
	cout << ret << endl;
	return 0;
}

File: PS_Python/가운데_글자_가져오기.py
def solution(s):

    s_len = len(s)
    if s_len == 1 or s_len == 2:
        return s
    answer = ""

    half_s_len = s_len // 2

    if s_len % 2 == 0:
        return s[half_s_len - 1] + s[half_s_len]
    else:
        return s[half_s_len]


print(solution("qwer"))
